src/OpenTelemetry.AutoInstrumentation/Instrumentations/MongoDb/MongoClientIntegration.cs

```diff
@@ -0,0 +1,151 @@
+// <copyright file="MongoClientIntegration.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using OpenTelemetry.AutoInstrumentation.CallTarget;
+
+namespace OpenTelemetry.AutoInstrumentation.Instrumentations.MongoDb
+{
+    /// <summary>
+    /// MongoDB.Driver.MongoClient calltarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "MongoDB.Driver",
+        TypeName = "MongoDB.Driver.MongoClient",
+        MethodName = ".ctor",
+        ReturnTypeName = ClrNames.Void,
+        ParameterTypeNames = new[] { "MongoDB.Driver.MongoClientSettings" },
+        MinimumVersion = "2.3.0",
+        MaximumVersion = "2.65535.65535",
+        IntegrationName = "MongoDb")]
+    public class MongoClientIntegration
+    {
+#if NETCOREAPP3_1_OR_GREATER
+        private static Delegate _setActivityListener;
+#endif
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TMongoClientSettings">Type of the settings</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="settings">The source of the original GraphQL query</param>
+        /// <returns>Calltarget state value</returns>
+        public static CallTargetState OnMethodBegin<TTarget, TMongoClientSettings>(TTarget instance, TMongoClientSettings settings)
+        {
+#if NETFRAMEWORK
+            // Additional deps doesn't support .NET FX
+            // TODO: Find another way how to ship "MongoDB.Driver.Core.Extensions.DiagnosticSources"
+            return CallTargetState.GetDefault();
+#endif
+
+#if NETCOREAPP3_1_OR_GREATER
+            Assembly driverExtensions;
+            Type driverAcitivitySubscriber = Type.GetType("MongoDB.Driver.Core.Extensions.DiagnosticSources.DiagnosticsActivityEventSubscriber, MongoDB.Driver.Core.Extensions.DiagnosticSources");
+            if (driverAcitivitySubscriber == null)
```

Comment:
In your testing, did you make sure that this if statement is only entered once? If not, I think you might need to assign `driverAcitivitySubscriber = driverExtensions.GetType("MongoDB.Driver.Core.Extensions.DiagnosticSources.DiagnosticsActivityEventSubscriber")`

Reply:
Seems the else part of the if is never invoked 🤔

Reply:
solved the original issue. https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/421#pullrequestreview-925911047